system/jlib/jbuff.cpp

```diff
@@ -554,6 +554,18 @@ MemoryBuffer & MemoryBuffer::append(unsigned __int64 value)
     return appendEndian(sizeof(value), &value);
 }
 
+MemoryBuffer & MemoryBuffer::appendPacked(unsigned __int64 value)
+{
+    //Append bytes with the top bit set until the value is less than 0x80
+    while (value >= 0x80)
+    {
+        byte next = ((byte)value) | 0x80;
+        append(next);
+        value >>= 7;
+    }
+    return append((byte)value);
```

Comment:
Just checking, but worst case here is to append 10 bytes?
But approach makes sense where usually relatively small numbers are being serialized.